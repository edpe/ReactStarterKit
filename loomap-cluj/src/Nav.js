import React from 'react';

const Nav = (props) => {
  return (
    <nav>
      <button>List</button>
      <button>Map</button>
    </nav>
  );
}

export default Nav;
