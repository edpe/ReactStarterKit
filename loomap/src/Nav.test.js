import React from 'react';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';

import Nav from './Nav';

describe('Nav', () => {
  it('Matches the snapshot', () => {
    const component = mount(<Nav/>);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Calls showList when the list button is clicked', () => {
    const showList = jest.fn();
    const component = mount(<Nav showList={showList}/>);
    component.find('button').at(0).simulate('click');
    expect(showList).toHaveBeenCalled();
  });

  it('Calls showMap when the map button is clicked', () => {
    const showMap = jest.fn();
    const component = mount(<Nav showMap={showMap}/>);
    component.find('button').at(1).simulate('click');
    expect(showMap).toHaveBeenCalled();
  });
});
