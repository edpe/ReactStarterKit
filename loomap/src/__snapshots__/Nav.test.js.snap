// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Matches the snapshot 1`] = `
<Nav>
  <nav>
    <button>
      List
    </button>
    <button>
      Map
    </button>
  </nav>
</Nav>
`;
